zero_functional.nim

```diff
@@ -350,7 +382,7 @@ proc inlineIndex(ext: ExtNimNode): ExtNimNode{.compileTime.} =
 proc inlineFold(ext: ExtNimNode): ExtNimNode{.compileTime.} =
   let initialValue = ext.node[1]
   let resultIdent = ext.res
-  let foldOperation = ext.adapt(index=2)
+  let foldOperation = ext.adapt(index=2, inFold=true)
```

Comment:
I had some problems when naming the list `a` and refering to `a` inside non-fold operations - so I added the check if we are in a fold and then in apdapt change the variable name to __a__ - otherwise a few of the tests will stop running...